compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -1348,33 +1348,47 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         span: Span,
         remaining_fields: FxHashMap<Ident, (usize, &ty::FieldDef)>,
     ) {
-        let tcx = self.tcx;
         let len = remaining_fields.len();
 
         let mut displayable_field_names =
             remaining_fields.keys().map(|ident| ident.as_str()).collect::<Vec<_>>();
 
         displayable_field_names.sort();
 
-        let truncated_fields_error = if len <= 3 {
-            String::new()
+        let truncated_fields_error: String;
+        let remaining_fields_names = if len <= 3 {
+            truncated_fields_error = String::new();
+
+            match len {
+                1 => format!("`{}`", displayable_field_names[0]),
+                2 => {
+                    format!("`{}` and `{}`", displayable_field_names[0], displayable_field_names[1])
+                }
+                3 => format!(
+                    "`{}`, `{}` and `{}`",
+                    displayable_field_names[0],
+                    displayable_field_names[1],
+                    displayable_field_names[2]
+                ),
+                _ => unreachable!(),
+            }
         } else {
-            format!(" and {} other field{}", (len - 3), if len - 3 == 1 { "" } else { "s" })
-        };
+            truncated_fields_error = format!(" and {} other field{}", len - 3, pluralize!(len - 3));
```

Comment:
Is there a test for this case?

Reply:
Yes, for example here: https://github.com/rust-lang/rust/blob/0d97f7a96877a96015d70ece41ad08bb7af12377/src/test/ui/error-codes/E0063.stderr#L19